security/RFP.md

```diff
@@ -0,0 +1,254 @@
+
+### Hydra plutus scripts are immune to common Cardano smart-contract weaknesses
+
+Evaluate the Hydra plutus scripts susceptibility to common possible vulnerabilities for Cardano smart contracts, such as but not limited to the following attacks:
+* Execution cost limits;
+* Double satisfaction;
+* Replay;
+* Denial of service.
+
+See, for example, https://github.com/Plutonomicon/plutonomicon/blob/main/vulnerabilities.md.
+
+See the draft https://plutus--4604.org.readthedocs.build/en/4604/reference/common-weaknesses/index.html.
+
+### Hydra node chain layer code generates transactions which are consistent with Hydra Head v1 specification
+
+The Coordinated Hydra Head V1 specification defines the transactions the off-chain code should build and post to evolve the head status on-chain.
+
+You will review the corresponding artifacts described above to assess that the Hydra node chain layer code can only build transactions which are consistent with the Coordinated Hydra Head V1 specification.
+
+### Hydra node logic layer code is consistent with Hydra Head v1 specification
+
+The Coordinated Hydra Head V1 specification defines the off-chain logic of a hydra head to create, maintain and close a head. The Hydra node logic layer code implements this off-chain logic.
+
+You will review the corresponding artifacts described above to assess that the Hydra node logic layer code faithfully implements the off-chain logic described in the Coordinated Hydra Head V1 specification. The outcome of the review should include, bo not being limited to:
+* identify any discrepancy with the off-chain logic from the specification;
+* for any such discrepancy, what security property could be impacted;
+* a review of the tests applied to the Hydra node logic layer and, in particular, any situation that would not be covered by the tests.
+
+This review should consider fair management of the input and output to and from the Hydra node logic layer.
+
+### Hydra head protocol implementation is immune to attacks via chain transactions
+
+The Hydra node chain layer is in charge of observing transactions posted on the Cardano blockchain
+and filtering _interesting_ transaction to submit them to the logic layer. The hydra node logic layer
+then analyse those transactions and react to them to make a head status evolve.
+
+You will review the Hydra node chain and logic layers and the corresponding artifacts in presence of
+invalid, unexpected or maliciously forged transactions. The outcome of the review should include, but not being limited to:
+* identification of transactions or a suite of transactions that would mislead a hydra node to an incorrect head state
+* identification of transactions or a suite of transactions that would lead to a denial of service for the hydra node
+* identification of transactions or a suite of transactions that would invalidate one of the security properties of the Coordinated Hydra Head V1 specification.
```

Comment:
Can't we do that on the specification itself? If we audit the spec and it's fine, we "only" need to audit the chain layer to be consistent with the assumptions from the paper?

Also applies to the network task below.

Reply:
Good point. Maybe I should add something for the spec V1 audit part about that.

I feel like, however, that it could be interesting to still check: is our actual code immune to actual on chain invalid transactions?

I don't know 🤔